src/redux/reducer.js

```diff
@@ -0,0 +1,25 @@
+import { REQUEST_SPENDINGS_SUCCESS } from './types';
+
+const initialState = {
+  spendings: [],
+  pagination: {
+    total: undefined,
+    page: 1,
+    perPage: 5,
+  },
+};
+
+const reducer = (state = initialState, action) => {
+  switch (action.type) {
+    case REQUEST_SPENDINGS_SUCCESS:
+      return {
+        ...state,
+        spendings: [...action.spendings],
+        pagination: { ...action.pagination },
```

Comment:
If `spendings` is already an array, why do you use spreading here? The same for `pagination`.